scripts/validate_data.py

```diff
@@ -21,54 +21,116 @@ def file_hash(filename):
-        'This is just a template -- you are expected to code this.')
+    return sha1(byte_contents).hexdigest()
 
 
-def validate_data(data_directory):
-    """ Read ``data_hashes.txt`` file in `data_directory`, check hashes
+def validate_data(*data_directories):
+    """ Read ``hash_list.txt`` from `data_directory`, then check hashes.
+    Accepts more than one data_directory using *args.
 
     Parameters
     ----------
     data_directory : str
-        Directory containing data and ``data_hashes.txt`` file.
+        Directory containing data and ``hash_list.txt``.
 
     Returns
     -------
     None
 
     Raises
     ------
+    TypeError : 
+        If `data_directory` cannot be casted to pathlib.Path
+    ValueError :
+        If `data_directory` returns False to `data_directory.is_dir()`
+    ValueError :
+        If `data_directory / hash_list.txt` returns False to `is_file()`
     ValueError:
         If hash value for any file is different from hash value recorded in
-        ``data_hashes.txt`` file.
+        ``hash_list.txt`` file.
     """
-    # Read lines from ``data_hashes.txt`` file.
-    # Split into SHA1 hash and filename
-    # Calculate actual hash for given filename.
-    # If hash for filename is not the same as the one in the file, raise
-    # ValueError
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError(
-        'This is just a template -- fill out the template with code.')
 
+    for data_directory in data_directories:
+        # Try data_directory as Path
+        try:
+            data_directory = Path(data_directory)
+        except TypeError:
+            raise TypeError(
+                "data_directory argument must be a pathlib.Path (or a type that supports"
+                " casting to pathlib.Path, such as string)."
+            )
+
+        # Get absolute filename, expand ~user formats and resolving symlinks or relative paths
+        data_directory = data_directory.expanduser().resolve()
+
+        # Check if directory exists
+        if not data_directory.is_dir():
+            raise ValueError(f"Directory not found: {data_directory}.")
+        
+        # Get hash file path
+        hash_path = data_directory / "hash_list.txt"
+
+        # Check if any hash file exist
+        if not hash_path.is_file():
+            raise ValueError(f"File not found: {hash_path}.")
+
+        # Read lines from ``data_hashes.txt`` file.
+        hashes_lines = hash_path.read_text().splitlines()
+        for line in hashes_lines:
+            # Split into SHA1 hash and filename
+            expected_hash, filename = line.split()
+
+            # Calculate actual hash for given filename.
+            actual_hash = file_hash(data_directory.parent / filename)
+
+            # If hash for filename is not the same as the one in the file, raise
+            # ValueError
+            if not actual_hash == expected_hash:
+                raise ValueError(
```

Comment:
We should maybe raise a `RuntimeError` here. `ValueError` are generally raised for unexpected values of arguments.